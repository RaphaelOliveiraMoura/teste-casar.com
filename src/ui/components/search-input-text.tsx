"use client";

import { Search } from "lucide-react";
import { useRouter } from "next/navigation";

import { cn } from "../services/classname";
import { debounce } from "../services/debounce";

type SearchInputTextProps = React.InputHTMLAttributes<HTMLInputElement>;

export function SearchInputText({ className, ...props }: SearchInputTextProps) {
  const { replace } = useRouter();

  const handleSearchText = debounce((searchText: string) => {
    const urlParams = new URLSearchParams();
    urlParams.set("q", searchText);
    replace(`/detalhes?${urlParams.toString()}`);
  });

  return (
    <label className={cn(className, "relative w-full max-w-[600px]")}>
      <input
        className="w-full rounded-sm border py-2 pl-4 pr-8"
        type="text"
        onChange={(e) => handleSearchText(e.target.value)}
        {...props}
      />
      <div className="absolute right-2 top-1/2 -translate-y-1/2">
        <Search />
      </div>
    </label>
  );
}
