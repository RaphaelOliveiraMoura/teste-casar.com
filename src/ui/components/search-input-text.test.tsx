import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { useRouter as mockRouter } from "next/navigation";
import { expect, test } from "vitest";

import { SearchInputText } from "./search-input-text";

const { replace } = mockRouter();

test("<SearchInputText />", async () => {
  const snapshot = render(
    <SearchInputText placeholder="text input placeholder" />,
  );

  const input = screen.getByPlaceholderText("text input placeholder");
  expect(input).toBeDefined();

  fireEvent.change(input, { target: { value: "search text" } });

  await waitFor(() =>
    expect(replace).toBeCalledWith("/detalhes?q=search+text"),
  );

  expect(snapshot).toMatchSnapshot();
});
