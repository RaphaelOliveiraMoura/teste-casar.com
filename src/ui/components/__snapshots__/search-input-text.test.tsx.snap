// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SearchInputText /> 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="relative"
      >
        <input
          class="w-full rounded-sm border py-2 pl-4 pr-8"
          placeholder="text input placeholder"
          type="text"
          value=""
        />
        <div
          class="absolute right-2 top-1/2 -translate-y-1/2"
        >
          <svg
            class="lucide lucide-search"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <path
              d="m21 21-4.3-4.3"
            />
          </svg>
        </div>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="relative"
    >
      <input
        class="w-full rounded-sm border py-2 pl-4 pr-8"
        placeholder="text input placeholder"
        type="text"
        value=""
      />
      <div
        class="absolute right-2 top-1/2 -translate-y-1/2"
      >
        <svg
          class="lucide lucide-search"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            r="8"
          />
          <path
            d="m21 21-4.3-4.3"
          />
        </svg>
      </div>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
